### ISSUE TITLE
Addition of DP problem: Matrix Chain Multiplication [JAVA and CPP/C]

### ISSUE THREAD

**Issue by aryanargupta** [iiitv/algos#820]
#### This issue is about addition of a DP problem: [Matrix Chain Multiplication](https://www.geeksforgeeks.org/matrix-chain-multiplication-dp-8/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

**Comment by kalashshah** [comment]
Please read the [contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#suggesting-an-algorithm--ds) properly.

**Comment by aryanargupta** [comment]
> Please read the [contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#suggesting-an-algorithm--ds) properly.

Yes sir, I have read them properly. Is there anything specific to be more precise about in this issue?

**Comment by AkhileshManda** [comment]
@aryanargupta please read the first point that you checked while creating the issue

**Comment by aryanargupta** [comment]
> @aryanargupta please read the first point that you checked while creating the issue

Was the title of the issue the main problem? I have changed the title of the issue now.

**Comment by kalashshah** [comment]
GFI @aryanargupta

**Comment by aryanargupta** [comment]
> GFI @aryanargupta

Done @kalashshah sir 👍 Please review and suggest any changes if required
